evadb/executor/create_index_executor.py

```diff
@@ -33,26 +35,41 @@ def __init__(self, db: EvaDBDatabase, node: CreateIndexPlan):
         super().__init__(db, node)
 
     def exec(self, *args, **kwargs):
-        if self.catalog().get_index_catalog_entry_by_name(self.node.name):
-            msg = f"Index {self.node.name} already exists."
-            if self.node.if_not_exists:
-                logger.warn(msg)
-                return
-            else:
-                logger.error(msg)
-                raise ExecutorError(msg)
-
-        self.index_path = self._get_index_save_path()
-        self.index = None
-        self._create_index()
+        # Vector type specific creation.
+        if self.node.vector_store_type == VectorStoreType.PGVECTOR:
+            self._create_native_index()
+        else:
+            self._create_evadb_index()
 
         yield Batch(
             pd.DataFrame(
                 [f"Index {self.node.name} successfully added to the database."]
             )
         )
 
-    def _get_index_save_path(self) -> Path:
+    # Create index through the native storage engine.
+    def _create_native_index(self):
+        table = self.node.table_ref.table
+        db_catalog_entry = self.catalog().get_database_catalog_entry(
+            table.database_name
+        )
+
+        with get_database_handler(
+            db_catalog_entry.engine, **db_catalog_entry.params
+        ) as handler:
+            columns = table.table_obj.columns
+            # As other libraries, we default to HNSW and L2 distance.
```

Comment:
What does other libraries mean?

Reply:
I meant other vector store types that we currently support (e.g., FAISS).